edenia/antelope-tools
# Feat/array of endpoints 960

### Array of endpoints

### What does this PR do?

- Resolve #960

- To get the producers info an endpoint is requested, however if this endpoint fails the DB info doesn't change and bpjsons aren't updated, so now the functionality of get producers use a list of endpoints, each one is tried until the end of the list.
If this last case occurs the producers information is query from DB.

- The previous situation also happens in the website, the main page queries this endpoint a lot,  as some of them have a rate limit, when an endpoint fails it uses the next one in the list and wait 5 minutes before requesting the failed endpoint.

- Avoid requesting empty producers urls or urls shorter than 3 characters.
- Avoid to the delete nodes of default bpjson when is the eos network. 
- Now to get the server version the node api is requested after all the data is in the DB.


### Steps to test

1. Run the project locally (The first endpoint of the array must be failing to do the test)
2. Check the data in hasura (eos could be slow to get all data)
3. Go to frontend 
4. Check home page and "accounts and contracts" pages

Also check the order and endpoints that are used in the yamls files

#### Screenshots

Shows how when producers are loading it tests each endpoint until one works. (This logs are not in the pr)

![imagen](https://user-images.githubusercontent.com/66583677/199594721-d08d9022-4c13-40f0-b8ec-eab77429474e.png)

Frontend implementation:
![imagen](https://user-images.githubusercontent.com/66583677/199595142-79d30640-dd11-4942-9c4b-6fd2e6618f4a.png)
